src/Abp/Runtime/Caching/AbpCacheBase.cs

```diff
@@ -77,48 +77,48 @@ public virtual TValue Get(TKey key, Func<TKey, TValue> factory)
                 }
             }
 
-            return item;
+            return value;
         }
 
         public virtual TValue[] Get(TKey[] keys, Func<TKey, TValue> factory)
         {
-            TValue[] items = null;
+            TValue[] values = null;
 
             try
             {
-                items = GetOrDefault(keys);
+                values = GetOrDefault(keys);
             }
             catch (Exception ex)
             {
                 Logger.Error(ex.ToString(), ex);
             }
 
-            if (items == null)
+            if (values == null)
             {
-                items = new TValue[keys.Length];
+                values = new TValue[keys.Length];
             }
 
-            if (items.Any(i => i == null))
+            if (values.Any(v => EqualityComparer<TValue>.Default.Equals(v, default)))
```

Comment:
Your previous implementation introduced the bug that, for `int`, `0` was not stored.
You current implementation introduces the bug that, for `int`, `0` is stored but ignored.
In all implementations, including the existing one and my suggestions, `null` is either not stored or ignored.

I suggest that, instead of going with a half-baked solution, we address the original issue first using one of the implementations in https://github.com/aspnetboilerplate/aspnetboilerplate/pull/5277#discussion_r377135879 that does not introduce a new bug.

Reply:
I came out with the changes for `TryGetValue()` in https://github.com/aspnetboilerplate/aspnetboilerplate/pull/5334.

I will create a separate PR for the equality problem.

Reply:
@acjh I removed all the other changes out of this PR. 😄